src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -601,7 +601,12 @@ public override bool Equals(object obj)
 
         public bool Equals(ISymbol other)
```

Comment:
>public bool Equals(ISymbol other) [](start = 8, length = 33)

Are there still callers to this overload? If not, consider marking as `[Obsolete]`.

Reply:
Yeah, I thought about this, and actually I agree that we *should* make this obsolete, as it is no longer really sufficient to compare types with.

Unfortunately *lots* of analyzers call this method (which is why we need to do the work in the first place), so we'd add a lot of warnings to them. For now I think we should open an issue to track it, but leave it as-is for this release.

Reply:
this really concerns me.  it seems like we're just going to be putting analyzers in a bad place anyways since their checks will be wrong.  So why not acutally let them know about it instead of having it silently seem like it's succeeding.

it's not like we're breaking binary compat.  we're just makign it clear where they need to fix things up to make their code correct.  otherwise, they'll just be flying blind here.  

tagging @333fred as these conversations came up with the entire nullable API surface area.

Reply:
@CyrusNajmabadi Most of the time, the checks will probably be correct. This is providing a mechanism for incremental adoption of nullability. Otherwise even simple things like asking if a type is `List<string>` will be wrong because the nullability may not match.